### DISCUSSION TITLE
Is this incompatible with Qwik?

### DISCUSSION THREAD

**Discussion by UpToNathan** [WJSoftware/vite-plugin-single-spa#143] [Q&A] [answered]
So I've had a few unsuccessful attempts at getting this to work with Qwik, wondered if anyone else has tried as well?

**Comment by webJose** [comment]
Hello.  I don't even know what Qwik is, so I guess the probability of this plug-in being incompatible with it is high.  If you provide information on Qwik, I could evaluate if it is something that I would like to pursue as part of this plug-in's roadmap.  Cheers.

**Comment by webJose** [comment, marked as answer]
I got curious and quickly searched for Qwik.  It is just another framework, like React or Svelte.  If Vite can bundle it, and you can provide single-spa lifecycle functions, this plug-in will do the rest.

The problem:  There is no `single-spa-qwik` package available.  Someone versed in how Qwik instantiates components should be able to create said package.

**Reply by UpToNathan** [reply]
Hey thanks for replying so soon! So I hacked a single-spa-qwik together, kind of, I think. That said, when I add this plugin to vite config and try to build I get an error saying 'input.startsWith is not a function'. Got a bit stuck trying to figure out why though

**Reply by webJose** [reply]
I just searched the entire code base of `vite-plugin-single-spa` for "startswith" and found no matches, so I can only imagine the error is coming from your attempt of creating lifecycle functions.  Maybe you should seek help from the Qwik community.

**Reply by UpToNathan** [reply]
Sure, thanks all the same for checking
